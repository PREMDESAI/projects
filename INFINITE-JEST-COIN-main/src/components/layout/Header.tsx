import MarqueeText from './MarqueeText';
import WalletButton from '../ui/WalletButton';

const Header = () => {
    return (
        <div className="fixed top-0 z-50 w-full whitespace-nowrap overflow-hidden border-b border-neutral-200 bg-white bg-opacity-90 backdrop-blur-sm">
            <div className="relative flex items-center justify-between px-4 transform-gpu perspective-1000">
                <div className="flex-1 overflow-hidden">
                    <MarqueeText />
                    <MarqueeText isSecondary />
                </div>
                <div className="flex-shrink-0 ml-4">
                    <WalletButton />
                </div>
            </div>
        </div>
    );
};

export default Header; 