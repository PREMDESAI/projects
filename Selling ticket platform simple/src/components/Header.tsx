import { useState } from 'react';
import { Menu, X } from 'lucide-react';

const Header = () => {
  const [isMenuOpen, setIsMenuOpen] = useState(false);

  return (
    <header className="fixed w-full bg-black/80 backdrop-blur-md z-50">
      <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
        <div className="flex justify-between items-center h-20">
          {/* Logo */}
          <div className="flex-shrink-0">
            <a href='#' className="text-3xl font-bold bg-gradient-to-r from-red-500 to-blue-500 bg-clip-text text-transparent">
              SOUNDUOEX
            </a>
          </div>

          {/* Desktop Navigation */}
          <nav className="hidden md:flex space-x-8">
            {['About', 'Artists', 'Merchandise', 'Digital', 'FAQ', 'Contact'].map((item) => (
              <a
                key={item}
                href={`#${item.toLowerCase()}`}
                className="text-gray-300 hover:text-white transition-colors duration-200"
              >
                {item}
              </a>
            ))}
          </nav>

          {/* CTA Button */}
          <div className="hidden md:block">
            <button className="bg-purple-600 px-6 py-2 rounded-full hover:bg-purple-700 transition">
              Buy Tickets
            </button>
          </div>

          {/* Mobile Menu Button */}
          <div className="md:hidden">
            <button
              onClick={() => setIsMenuOpen(!isMenuOpen)}
              className="text-gray-300 hover:text-white"
            >
              {isMenuOpen ? <X size={24} /> : <Menu size={24} />}
            </button>
          </div>
        </div>

        {/* Mobile Menu */}
        {isMenuOpen && (
          <div className="md:hidden">
            <div className="px-2 pt-2 pb-3 space-y-1">
              {['About', 'Artists', 'Merchandise', 'Digital', 'FAQ', 'Contact'].map((item) => (
                <a
                  key={item}
                  href={`#${item.toLowerCase()}`}
                  className="block px-3 py-2 text-gray-300 hover:text-white"
                  onClick={() => setIsMenuOpen(false)}
                >
                  {item}
                </a>
              ))}
              <button className="w-full mt-4 bg-gradient-to-r from-red-500 to-blue-500 text-white px-6 py-2 rounded-full hover:opacity-90 transition-opacity duration-200">
                Buy Tickets
              </button>
            </div>
          </div>
        )}
      </div>
    </header>
  );
};

export default Header;