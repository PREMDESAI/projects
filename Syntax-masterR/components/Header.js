import Subscribe from './Subscribe';
import styled from 'styled-components';
import Link from 'next/link';
import { theme, media, StyledHeader } from '../styles';
const { colors } = theme;

const HeaderContainer = styled(StyledHeader)`
  display: flex;
  flex-wrap: wrap;
  margin: 2rem auto;
  color: ${colors.white};
`;
const HeaderLeft = styled.div`
  width: 30%;
  text-align: center;
  ${media.tablet`
    width: 100%;
  `};
`;
const Logo = styled.img`
  margin-left: -3rem;
  max-width: 300px;
  text-align: center;
  ${media.tablet`
    margin-left: -2rem;
  `};
`;
const HeaderRight = styled.div`
  width: 70%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  ${media.tablet`
    width: 100%;
  `};
`;
const Title = styled.div`
  position: relative;
`;
const Tagline = styled.h2`
  font-size: 2.5rem;
  margin: 0;
  ${media.desktop`
    text-align: center;
  `};
  ${media.tablet`;
    font-size: 1.5rem;
  `};
`;
const PotluckButton = styled.a`
  position: absolute;
  top: 5px;
  right: 0;
  padding: 5px 10px;
  border: 1px solid ${colors.yellow};
  border-radius: 3px;
  font-size: 12px;
  text-decoration: none;
  &:hover {
    border: 1px dotted;
  }
  ${media.desktop`
    position: relative;
    margin: 10px 0;
    text-align: center;
    display: block;
    top: 0;
  `};
`;
const People = styled.div`
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  margin: 1rem 0;
  font-size: 1.4rem;
`;
const Person = styled.div`
  background: rgba(255, 255, 255, 0.07);
  padding: 1rem;
  width: 48%;
  ${media.tablet`
    width: 100%;
    margin-bottom: 1rem;
  `};
`;
const Avatar = styled.img`
  border-radius: 50%;
  width: 80px;
  height: 80px;
  margin-right: 20px;
  border: 3px solid ${colors.white};
  box-shadow: inset 0 0 10px red;
  ${media.tablet`
    width: 50px;
    height: 50px;
    border-width: 1px;
  `};
`;
const Name = styled.h3`
  margin: 0;
  font-size: 2rem;
  em {
    font-size: 1rem;
  }
`;
const PersonName = styled.div`
  display: flex;
  align-items: center;
`;
const PersonDetails = styled.p`
  margin-bottom: 0;
`;
const Header = () => (
  <HeaderContainer>
    <HeaderLeft>
    <Link href="/">
      <a aria-label="Syntax.FM">
      <Logo src="/static/logo.png" alt="Syntax" />
      </a>
    </Link>
    </HeaderLeft>
    <HeaderRight>
      <Title>
        <Tagline>A Tasty Treats Podcast for Web Developers.</Tagline>
        <PotluckButton
          target="_blank"
          rel="noopener noreferrer"
          href="https://docs.google.com/forms/d/e/1FAIpQLSfQlAo1wXHiJMySdU-h8QMtfoz92aMS9eycEHXB6eRCLh8KHA/viewform"
        >
          Ask a Potluck Question â†’
        </PotluckButton>
      </Title>
      <People>
        <Person>
          <PersonName>
            <Avatar src="/static/wes400x400.jpg" alt="Wes Bos" />
            <div>
              <Name>Wes Bos</Name>
              <a
                target="_blank"
                rel="noopener noreferrer"
                href="https://twitter.com/wesbos"
              >
                @wesbos
              </a>
            </div>
          </PersonName>
          <PersonDetails>
            Full Stack JavaScript Developer. Creator of really good{' '}
            <a
              target="_blank"
              rel="noopener noreferrer"
              href="https://wesbos.com/courses"
            >
              web development courses
            </a>
            . BBQ enthusiast.
          </PersonDetails>
        </Person>
        <Person>
          <PersonName>
            <Avatar
              src="https://avatars2.githubusercontent.com/u/669383?s=460&v=4"
              alt="Scott Tolinski"
            />
            <div>
              <Name>Scott Tolinski</Name>
              <a
                target="_blank"
                rel="noopener noreferrer"
                href="https://twitter.com/stolinski"
              >
                @stolinski
              </a>
            </div>
          </PersonName>
          <PersonDetails>
            Web Developer, Creator of{' '}
            <a href="https://leveluptutorials.com/">Level Up Tuts</a>, Bboy,
            Robotops Crew and{' '}
            <a
              target="_blank"
              rel="noopener noreferrer"
              href="https://www.youtube.com/c/leveluptuts"
            >
              Youtuber
            </a>
          </PersonDetails>
        </Person>
      </People>
    </HeaderRight>
    <Subscribe />
  </HeaderContainer>
);

export default Header;
