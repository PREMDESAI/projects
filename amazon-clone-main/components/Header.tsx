"use client"
import Image from 'next/image'
import React, { useEffect, useState } from 'react'
import amazonLogo from "../public/amazon-logo-2.webp";
import { BiCart } from "react-icons/bi";
import { CgSearch } from "react-icons/cg";
import Link from 'next/link';
import { useRouter } from 'next/navigation';
import { useAppSelector } from '@/lib/supabase/hooks/redux';
import { getCart } from '@/redux/cartSlice';
import { supabase } from '@/lib/supabase/products';

const itemList = [
    "All",
    "Fresh",
    "Amazon miniTV",
    "Sell",
    "Gift Cards",
    "Baby",
    "Buy Again",
    "Browsing History",
    "Amazon Pay",
    "Gift Ideas",
    "Health, Household & Personal Care"
]

const Header = () => {
    const [query, setQuery] = useState<string>("");
    const [user, setUser] = useState<any>(null);
    const router = useRouter();


    const cart = useAppSelector(getCart);

    const searchHandler = () => {
        router.push(`/search/${query}`);
    }

    useEffect(()=>{
        const getUserData = async () => {
            const {data:{user}} = await supabase.auth.getUser();
            setUser(user);
        }
        getUserData();
    },[])
 
    return (
        <>
            <div className='bg-[#131921] text-white py-1'>
                <div className='flex items-center justify-between w-[90%] mx-auto'>
                    <Link href={'/'} className='w-[10%]'>
                        <Image src={amazonLogo} alt={"logo"} width={150} height={150} />
                    </Link>
                    <div className='flex items-center w-[60%]'>
                        <input 
                        value={query} 
                        onChange={(e)=>setQuery(e.target.value)}
                        type="text" 
                        className='w-full p-2 rounded-l-md outline-none text-black' placeholder='Search Amazon.in' />
                        <div 
                        onClick={searchHandler}
                        className='bg-[#FEBD69] p-2 cursor-pointer hover:bg-[#ffad43]   rounded-r-md'>
                            <CgSearch size={"24px"} className='text-black' />
                        </div>
                    </div>
                    <div className='flex items-center justify-around w-[20%]'>
                        <div onClick={()=>{
                            router.push("/signin")
                        }} className='cursor-pointer'>
                            <h1 className='text-xs hover:underline'>{`${user ? user?.identities[0]?.identity_data.full_name:"Signin"}`}</h1>
                            <h1 className='font-medium text-sm'>Account & Lists</h1>
                        </div>
                        <div>
                            <p className='text-xs'>Returns</p>
                            <h1 className='font-medium text-sm'>& Orders</h1>
                        </div>
                        <Link href={"/cart"}  className='cursor-pointer'>
                            <p className='relative top-3 left-5'>{cart.length}</p>
                            <div className='flex'>
                                <div>
                                    <BiCart size={"40px"} />
                                </div>
                                <h1 className='mt-4'>cart</h1>
                            </div>
                        </Link>
                    </div>
                </div>
            </div>
            <div className='bg-[#232F3E] w-full text-white p-2  flex justify-between items-center'>
                <div>
                    {
                        itemList.map((link, idx) => {
                            return (
                                <Link key={idx} href={`/${link}`} className='mx-2 hover:border border border-transparent hover:border-white p-2'>
                                    {link}
                                </Link>
                            )
                        })
                    }

                </div>
                <div className='mr-5'>
                    <h1 onClick={ async ()=>{
                        const {error} = await supabase.auth.signOut();
                        router.push("/signin");
                    }} className='text-[#FEBD69] font-bold cursor-pointer hover:underline' >Sign out</h1>
                </div>
            </div>

        </>
    )
}

export default Header