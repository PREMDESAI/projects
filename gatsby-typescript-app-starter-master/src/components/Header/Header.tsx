import Link from "gatsby-link"
import React, { FunctionComponent } from "react"

interface Props {
  siteTitle: string
}

const Header: FunctionComponent<Props> = ({ siteTitle = "" }: Props) => (
  <header
    style={{
      background: `rebeccapurple`,
      marginBottom: `1.45rem`,
    }}
  >
    <div
      style={{
        margin: `0 auto`,
        maxWidth: 960,
        padding: `1.45rem 1.0875rem`,
      }}
    >
      <h1 style={{ margin: 0 }}>
        <Link
          to="/"
          style={{
            color: `white`,
            textDecoration: `none`,
          }}
        >
          {siteTitle}
        </Link>
      </h1>
    </div>
  </header>
)

export default Header
