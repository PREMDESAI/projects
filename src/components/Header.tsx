import React from 'react';

interface IProps {
    title: string;
    currency: string;
    totalMin: number;
    totalMax: number;
    onEstimateToggle: () => void;
}

export class Header extends React.PureComponent<IProps> {
    public render() {
        const { currency, totalMin, totalMax } = this.props;
        return (
            <div className='navbar navbar-dark bg-dark shadow-sm'>
                <div className='container d-flex justify-content-between'>
                    <div className='navbar-brand d-flex align-items-center w-100 mr-0'>
                        <div className='d-flex justify-content-between w-100'>
                            <div className='d-none d-lg-block'>{ this.props.title }</div>
                            <div className='d-lg-none d-sm-block d-md-block'>Calculator</div>
                            <button type='button' className='btn btn-outline-light d-lg-none d-sm-block d-md-block btn-sm' onClick={this.props.onEstimateToggle}>
                                { currency }{ totalMin } - { currency }{ totalMax }
                            </button>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}