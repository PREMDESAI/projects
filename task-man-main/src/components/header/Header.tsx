import { useTasks } from '@/hooks/use-tasks';
import { GearIcon } from '@radix-ui/react-icons';
import { Flex, Heading } from '@radix-ui/themes';
import { keyframes, styled } from '@stitches/react';

const spinner = keyframes({
  '0%': { transform: 'rotate(0deg)' },
  '100%': { transform: 'rotate(45deg)' },
});

const LoadingIcon = styled(GearIcon, {
  animation: `${spinner} 1s infinite`,
});

export const Header = () => {
  const { isLoading } = useTasks();
  return (
    <Flex
      align='center'
      style={{
        width: '100vw',
        height: '64px',
        padding: '16px 32px',
        background: '#3E63DD',
        color: '#f5f5fa',
        position: 'sticky',
        top: 0,
      }}
    >
      <Heading>TASKMAN</Heading>
      {isLoading && (
        <LoadingIcon width='24' height='24' style={{ marginLeft: '8px' }} />
      )}
    </Flex>
  );
};
