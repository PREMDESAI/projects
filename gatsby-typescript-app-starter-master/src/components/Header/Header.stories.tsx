import React from "react"
import { storiesOf } from "@storybook/react"

import Header from "./Header"

const story = storiesOf("Components/Header", module)

story.add("Header", () => <Header siteTitle={"sample"} />)
