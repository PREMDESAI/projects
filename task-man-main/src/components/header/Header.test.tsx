import { render } from '@testing-library/react';
import { describe, it } from 'vitest';

import { Header } from './Header';

describe('Header', () => {
  it('renders', async () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
