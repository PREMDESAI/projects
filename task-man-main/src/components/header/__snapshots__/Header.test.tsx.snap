// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > renders 1`] = `
<div>
  <div
    class="rt-Flex rt-r-ai-center"
    style="width: 100vw; height: 64px; padding: 16px 32px; background: rgb(62, 99, 221); color: rgb(245, 245, 250); position: sticky; top: 0px;"
  >
    <h1
      class="rt-Heading rt-r-size-6"
    >
      TASKMAN
    </h1>
  </div>
</div>
`;
